diff -urN old/src/gpio.cpp new/src/gpio.cpp
--- old/src/gpio.cpp	2018-10-02 21:55:15.524457080 -0000
+++ new/src/gpio.cpp	2018-10-02 22:34:40.024500832 -0000
@@ -523,7 +523,7 @@
 	// =====================================================
 
 	ntx.modes = 0; /* only read */
-	status = ntp_adjtime(&ntx);
+	status = adjtimex(&ntx);
 	double ntp_ppm;
 
 	if (status != TIME_OK)
diff -urN old/src/mailbox.c new/src/mailbox.c
--- old/src/mailbox.c	2018-10-02 21:55:15.524457080 -0000
+++ new/src/mailbox.c	2018-10-03 00:42:44.794643028 -0000
@@ -37,6 +37,7 @@
 #include <sys/mman.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
+#include <sys/types.h>
 
 #include "mailbox.h"
 
