diff -urN old/src/rtl_eeprom.c new/src/rtl_eeprom.c
--- old/src/rtl_eeprom.c	2018-07-03 21:59:01.000000000 -0000
+++ new/src/rtl_eeprom.c	2018-10-07 22:29:01.312488122 -0000
@@ -370,14 +370,14 @@
 	}
 
 	if (manuf_str)
-		strncpy((char*)&conf.manufacturer, manuf_str, MAX_STR_SIZE);
+		strncpy((char*)&conf.manufacturer, manuf_str, MAX_STR_SIZE - 1);
 
 	if (product_str)
-		strncpy((char*)&conf.product, product_str, MAX_STR_SIZE);
+		strncpy((char*)&conf.product, product_str, MAX_STR_SIZE - 1);
 
 	if (serial_str) {
 		conf.have_serial = 1;
-		strncpy((char*)&conf.serial, serial_str, MAX_STR_SIZE);
+		strncpy((char*)&conf.serial, serial_str, MAX_STR_SIZE - 1);
 	}
 
 	if (ir_endpoint != 0)
